### PR TITLE
chore: update examples with new createVirtualEnvironment(shape, {...}) api

### PR DESCRIPTION
**Pull request by rwaldron**
Depends on #204

### CODE REVIEW

File: examples/errors/index.js [hunk 1]
```diff
@@ -20,7 +24,9 @@ try {
     `);
 } catch (e) {
     // testing syntax error when evaluating
-    e instanceof SyntaxError;
+    if (!(e instanceof SyntaxError)) {
+        console.log(e);
```
Comment (jdalton) [inline examples/errors/index.js:28]: `console.error`? ☝️

File: examples/errors/index.js [hunk 2]
```diff
@@ -29,7 +35,9 @@ try {
     `);
 } catch (e) {
     // testing initialization error when evaluating
-    e instanceof Error;
+    if (!(e instanceof Error)) {
+        console.log(e);
```
Comment (jdalton) [inline examples/errors/index.js:39]: ditto?

File: examples/expandos/index.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import createVirtualEnvironment from '@locker/near-membrane-dom';
 
-const evalScript = createVirtualEnvironment();
+const evalScript = createVirtualEnvironment(window, { globalThis: window });
```
Comment (jdalton) [inline examples/expandos/index.js:3]: Could `globalThis` default to the first argument?

File: examples/custom-elements/index.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import createVirtualEnvironment from '@locker/near-membrane-dom';
 
-const evalScript = createVirtualEnvironment();
-
+const evalScript = createVirtualEnvironment(window, {
+    globalThis: window,
```
Comment (caridy) [inline examples/custom-elements/index.js:4]: this is the default behavior anyways.

File: examples/errors/index.js [hunk 5]
```diff
@@ -1,17 +1,21 @@
...
+    ],
 ]);
 
 globalThis.bar = { a: 1, b: 2 };
-Object.freeze(globalThis.bar)
+Object.freeze(globalThis.bar);
 
-const evalScript = createVirtualEnvironment({
+const evalScript = createVirtualEnvironment(window, {
     distortionMap,
```
Comment (caridy) [inline examples/errors/index.js:16]: this is incorrect, distortionMap is not longer a valid option, but a callback

File: examples/errors/index.js [hunk 6]
```diff
@@ -1,17 +1,21 @@
...
-    endowments: window
+    endowments: globalThis,
```
Comment (caridy) [inline examples/errors/index.js:17]: unnecessary since you are already providing that as shape.

File: examples/errors/index.js [hunk 7]
```diff
@@ -1,17 +1,21 @@
...
+    globalThis: window,
```
Comment (caridy) [inline examples/errors/index.js:18]: default behavior, unnecessary.

File: examples/expandos/index.js (hunk 3 above)
Comment (caridy) [reply examples/expandos/index.js:3]: it doesn't default to the first argument, but to the current globalThis.

Comment (caridy) [reply examples/expandos/index.js:3]: I'm getting more uncomfortable with the default behavior, this is a low level library, maybe we should just remove all those default behaviors, in which case consumers of this library should pass the specific values for all 3

File: examples/getter-distortion/index.js [hunk 8]
```diff
@@ -4,13 +4,16 @@ import createVirtualEnvironment from '@locker/near-membrane-dom';
...
+        get,
+        () => {
+            console.error('forbidden');
+            return null;
+        },
+    ],
 ]);
 
-const evalScript = createVirtualEnvironment({ distortionMap });
+const evalScript = createVirtualEnvironment(window, { distortionMap, globalThis: window });
```
Comment (caridy) [inline examples/getter-distortion/index.js:16]: same here,,, distortionMap is not longer good, and globalThis is the default behavior

File: examples/invalid-fetch/index.js [hunk 9]
```diff
@@ -1,17 +1,24 @@
...
+    [
+        fetch,
+        () => {
+            console.error('forbidden');
+        },
+    ],
 ]);
-const evalScript = createVirtualEnvironment({
+const evalScript = createVirtualEnvironment(window, {
     distortionMap,
```
Comment (caridy) [inline examples/invalid-fetch/index.js:16]: same

File: examples/invalid-fetch/index.js [hunk 10]
```diff
@@ -1,17 +1,24 @@
...
-    endowments: window
+    endowments: {
+        originalFetch,
+        wrappedFetch,
+    },
+    globalThis: window,
```
Comment (caridy) [inline examples/invalid-fetch/index.js:21]: default

File: examples/object-graph-mutations/index.js [hunk 11]
```diff
@@ -1,6 +1,6 @@
 import createVirtualEnvironment from '@locker/near-membrane-dom';
 
-const evalScript = createVirtualEnvironment();
+const evalScript = createVirtualEnvironment(window, { globalThis: window });
```
Comment (caridy) [inline examples/object-graph-mutations/index.js:3]: globalthis is default behavior

File: packages/near-membrane-base/src/intrinsics.ts [hunk 12]
```diff
@@ -166,10 +166,12 @@ function getFilteredGlobalObjectShapeDescriptors(endowments: object): PropertyDe
 }
 
 export function getResolvedShapeDescriptors(
-    globalObjectShape: object,
+    globalObjectShape?: object,
```
Comment (caridy) [inline packages/near-membrane-base/src/intrinsics.ts:169]: I still don't understand why is this optional?

File: packages/near-membrane-base/src/intrinsics.ts [hunk 13]
```diff
@@ -166,10 +166,12 @@ function getFilteredGlobalObjectShapeDescriptors(endowments: object): PropertyDe
...
     endowments?: object
 ): PropertyDescriptorMap {
-    const globalObjectShapeDescriptors = getFilteredGlobalObjectShapeDescriptors(globalObjectShape);
+    const globalObjectShapeDescriptors = getFilteredGlobalObjectShapeDescriptors(
```
Comment (caridy) [inline packages/near-membrane-base/src/intrinsics.ts:172]: we should probably not even call the filter if the value is not defined

File: examples/expandos/index.js (hunk 3 above)
Comment (rwaldron) [reply examples/expandos/index.js:3]: @caridy that's certainly fine with me. I'll do it in a follow up

File: examples/custom-elements/index.js (hunk 4 above)
Comment (rwaldron) [reply examples/custom-elements/index.js:4]: After reading your comment [above](https://github.com/salesforce/near-membrane/pull/205#discussion_r721556000), about making things more explicit... I think we should further refine `createVirtualEnvironment`'s API to: 


```js
BrowserEnvironmentOptions {
    distortionCallback?: DistortionCallback;
    endowments?: object;
    keepAlive?: boolean;
    support?: SupportFlagsObject;
}

createVirtualEnvironment(
  globalObjectShape: object, 
  globalThis?: WindowProxy & typeof globalThis, 
  providedOptions?: BrowserEnvironmentOptions
)
```

File: packages/near-membrane-base/src/intrinsics.ts (hunk 12 above)
Comment (rwaldron) [reply packages/near-membrane-base/src/intrinsics.ts]: This was not intentional.

File: packages/near-membrane-base/src/intrinsics.ts (hunk 13 above)
Comment (rwaldron) [reply packages/near-membrane-base/src/intrinsics.ts:172]: This was from the other commit. We should discuss what we want to do if shape object is missing? Throw, or use an empty object?

**Pull request closed without merging**